packages/orm/src/collections/OneToOneReference.ts

```diff
@@ -79,7 +81,9 @@ export class OneToOneReference<T extends Entity, U extends Entity> extends Abstr
   }
 
   set(other: U): void {
-    ensureNotDeleted(this.entity);
+    if (this.entity.__orm.deleted) {
+      return;
```

Comment:
This should probably be `ensureNotDeleted(this.entity, { ignore: "pending" });` like it is in the other relations.

A good rule of thumb is to treat `__orm` as protected and not access it from outside an entity.  There are a handful of places where we aren't doing this, but we have been slowly removing those over time.

Reply:
Ah that makes sense.